packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -265,22 +265,19 @@ func BatchGetBlocksProofStatus(
 	defer cancel()
 	var (
 		parentHashes   = make([][32]byte, len(ids))
-		parents        = make([]*types.Header, len(ids))
-		blockIDs       = make([]uint64, len(ids))
+		parentIDs      = make([]*big.Int, len(ids))
+		blockIDs       = make([]*big.Int, len(ids))
+		blocks         = make([]uint64, len(ids))
 		result         = make([]*BlockProofStatus, len(ids))
 		highestBlockID = big.NewInt(0)
 	)
 	// Get the local L2 parent header.
-	g, gCtx := errgroup.WithContext(ctxWithTimeout)
+	g, _ := errgroup.WithContext(ctxWithTimeout)
 	for i, id := range ids {
 		g.Go(func() error {
-			parent, err := cli.L2.HeaderByNumber(gCtx, new(big.Int).Sub(id, common.Big1))
-			if err != nil {
-				return err
-			}
-			parentHashes[i] = parent.Hash()
-			parents[i] = parent
-			blockIDs[i] = id.Uint64()
+			parentIDs[i] = new(big.Int).Sub(id, common.Big1)
+			blockIDs[i] = id
+			blocks[i] = id.Uint64()
```

Comment:
This `blocks` var name is really confusing, let's rename it?

Reply:
OK